components/infobox/wikis/smash/infobox_league_custom.lua

```diff
@@ -0,0 +1,439 @@
+	Variables.varDefine('localcurrency', self.data.localCurrency)
+
+	-- Legacy date vars
+	Variables.varDefine('tournament_sdate', self.data.startDate)
+	Variables.varDefine('tournament_edate', self.data.endDate)
+	Variables.varDefine('tournament_date', self.data.endDate)
+	Variables.varDefine('formatted_tournament_date', self.data.startDate)
+	Variables.varDefine('date', self.data.endDate)
+	Variables.varDefine('sdate', self.data.startDate)
+	Variables.varDefine('edate', self.data.endDate)
+end
+
+--- @param args table
+--- @return string[]
+function CustomLeague:getWikiCategories(args)
+	local categories = {}
+
+	if args.game then
+		table.insert(categories, Game.name{game = args.game} .. ' Competitions')
+	end
+
+	if args.notranked then
+		table.insert(categories, 'Unranked Tournaments')
+	end
+
+	return categories
+end
+
+--- @param page string?
+--- @param query string
+--- @return series?
+function CustomLeague._querySeries(page, query)
+	if not page then
+		return
+	end
+
+	local sourcePagename = string.gsub(mw.ext.TeamLiquidIntegration.resolve_redirect(page), ' ', '_')
+
+	return mw.ext.LiquipediaDB.lpdb('series', {
+		conditions = '[[pagename::' .. sourcePagename .. ']]',
+		query = query,
+		limit = 1,
+	})[1]
+end
+
+--- @param page string?
+--- @return string?
+function CustomLeague.getAbbrFromSeries(page)
+	local series = CustomLeague._querySeries(page, 'abbreviation')
+	return series and series.abbreviation or nil
+end
+
+--- @param page string?
+--- @return string?
+--- @return string?
+function CustomLeague.getIconFromSeries(page)
+	local series = CustomLeague._querySeries(page, 'icon, icondark')
+	if not series then
+		return
+	end
+	return series.icon, series.icondark
+end
+
+--- @param content Html|string|number|nil
+--- @return string
+function CustomLeague._createNoWrappingSpan(content)
+	local span = mw.html.create('span')
+		:css('white-space', 'nowrap')
+		:node(content)
+	return tostring(span)
+end
+
+---@param fee string|number|nil
+---@param participants string|number|nil
+---@param bonus string|number|nil
+---@return number
+function CustomLeague._assumedPrize(fee, participants, bonus)
+	participants = tonumber(participants) or 0
+	fee = tonumber(fee) or 0
+	bonus = tonumber(bonus) or 0
+
+	return fee * participants + bonus
+end
+
+---@param localPrize string|number|nil
+---@param currency string
+---@param exchangeDate string?
+---@return number?
+function CustomLeague:_currencyConversion(localPrize, currency, exchangeDate)
+	local currencyRate = Currency.getExchangeRate{
+		currency = currency,
+		date = exchangeDate,
+		setVariables = true,
+	}
+	if not currencyRate then
+		return
+	end
+
+	return currencyRate * localPrize
+end
+
+--- Removes commas from a string
+---@param value string?
+---@return string?
+function CustomLeague._cleanNumericInput(value)
+	if Logic.isEmpty(value) then
+		return nil
+	end
+	---@cast value -nil
+
+	return (string.gsub(value, ',', ''))
+end
```

Comment:
would we maybe want to cast it to a number?
so we are sure it is numeric input we have
or do you want to allow text inputs too?

Reply:
Since many of the fields here (eg. prizepool) are normally not casted to numbers before being used by core code, I rather keep it that way instead of introducing two ways, increasing the risk for issues in the future